ddl/ddl_api.go

```diff
@@ -3547,6 +3547,11 @@ func needReorgToChange(origin *types.FieldType, to *types.FieldType) (needOreg b
 		toFlen, _ = mysql.GetDefaultFieldLengthAndDecimal(to.Tp)
 	}
 
+	if (types.IsTypeVarchar(origin.Tp) && to.Tp == mysql.TypeString) ||
+		(origin.Tp == mysql.TypeString && types.IsTypeVarchar(to.Tp)) {
```

Comment:
Same here, do we need any conversion going from char(N) to varchar(N<=M<256)?
Going from varchar() to char() needs conversion due to space padding to full lenght, which affects indexes, joins and other comparisons.